src/BlazorWebView/src/Maui/Windows/WinUIWebViewManager.cs

```diff
@@ -102,5 +115,30 @@ protected override void QueueBlazorStart()
 					");
 			};
 		}
+
+		private void Webview_CoreWebView2Initialized(WebView2Control sender, UI.Xaml.Controls.CoreWebView2InitializedEventArgs args)
+		{
+			_webview.CoreWebView2.NavigationStarting += CoreWebView2_NavigationStarting;
+			_webview.CoreWebView2.NewWindowRequested += CoreWebView2_NewWindowRequested;
+		}
+
+		private void CoreWebView2_NavigationStarting(CoreWebView2 sender, CoreWebView2NavigationStartingEventArgs args)
+		{
+			var uri = new Uri(args.Uri);
```

Comment:
Any chance these might not be valid? For e.g. should we do `TryCreate` and use LaunchUriAsync in the event we can't create it?

Reply:
Seems prudent, thanks!